templates/client.ts.hbs

```diff
@@ -12,6 +10,12 @@ import ResponseError from "./responseError";
 import TemplateURL from "./templateUrl";
 import { USER_AGENT_HEADER, USER_AGENT_VALUE } from "./version";
 
+if(isNode || !isBrowser || !hasFetchAvailable) {
+    const nodeFetch = require('node-fetch');
+    global.fetch = nodeFetch;
+    global.Request = nodeFetch.Request;
```

Comment:
@tarcang – these lines may have unintended side effects.

If you import `commerce-sdk-isomorphic`, it will overwrite `global.fetch` with the `node-fetch` implementation ... 

In Node 18+, `global.fetch` is something different, with a different interface.

Is there anyway to set `fetch` such that we don't overwrite the global?

Reply:
I was able to do it :)